docs/source/frameworks/tensorflow.rst

```diff
@@ -2,51 +2,236 @@
+                ]
+            )
+            def __call__(self, inputs):
+                return self.dense(inputs)
+
+
+        model = NativeModel()
+        ... # training
+
+        # =========
+
+        bentoml.tensorflow.save(model, "my_tf_model")
+
+   .. tab-item:: keras Sequential
+
+      .. code-block:: python
+
+        model = keras.models.Sequential(
+            (
+                keras.layers.Dense(
+                    units=1,
+                    input_shape=(5,),
+                    dtype=tf.float64,
+                    use_bias=False,
+                    kernel_initializer=keras.initializers.Ones(),
+                ),
+            )
+        )
+        opt = keras.optimizers.Adam(0.002, 0.5)
+        model.compile(optimizer=opt, loss="binary_crossentropy", metrics=["accuracy"])
+
+        bentoml.tensorflow.save(model, "my_keras_model")
+
+   .. tab-item:: keras Functional
+
+      .. code-block:: python
+    
+        x = keras.layers.Input((5,), dtype=tf.float64, name="x")
+        y = keras.layers.Dense(
+            6,
+            name="out",
+            kernel_initializer=keras.initializers.Ones(),
+        )(x)
+        model = keras.Model(inputs=x, outputs=y)
+
+        bentoml.tensorflow.save(model, "my_keras_model")
+
+
+Sometime a model may take multiple tensors as input
+
+.. code-block:: python
+
+   class MultiInputModel(tf.Module):
         def __init__(self):
-            super().__init__()
-            self.weights = np.asfarray([[1.0], [1.0], [1.0], [1.0], [1.0]])
-            self.dense = lambda inputs: tf.matmul(inputs, self.weights)
+            ...
 
         @tf.function(
-            input_signature=[tf.TensorSpec(shape=[1, 5], dtype=tf.float64, name="inputs")]
+            input_signature=[
+                tf.TensorSpec(shape=[1, 5], dtype=tf.float64, name="x1"),
+                tf.TensorSpec(shape=[1, 5], dtype=tf.float64, name="x2"),
+                tf.TensorSpec(shape=(), dtype=tf.float64, name="factor"),
+            ]
+        )
+        def __call__(self, x1: tf.Tensor, x2: tf.Tensor, factor: tf.Tensor):
+            ...
+
+    model = MultiInputModel()
+    ... # training
+
+    bentoml.tensorflow.save(model, "my_tf_model")
+
+.. seealso::
+   `bentoml.tensorflow.save`_ has two parameters `tf_signature` and `signatures`.
+   They are important when you want to save a model with a ensured behavior.
+   The `tf_signature` is a dict of tensor names and their shapes, which adaopted from
+   the signature of tensorflow.saved_model.save. You may find more details about it
+   on the `tensorflow.saved_model.save`_ documentation.
+   The `signatures` is a dict of functions names and some other information.
+   If you know your model has a dynamic batch dimension, you can use `signatures` to tell
+   bentoml about that for possible future optimizations like this:
+
+   bentoml.tensorflow.save(model, "my_model", signatures={"__call__": {"batch_dim": 0, "batchable": True}})
```

Comment:
Should `tf_signature` and `signatures` arguments be mutually exclusive? Let's also link the TF documentation here.